core/App/ehsm_provider.cpp

```diff
@@ -149,91 +149,53 @@ void Finalize()
     sgx_destroy_enclave(g_enclave_id);
 }
 
-static uint32_t sgx_calc_gcm_data_size(const uint32_t aad_size, const uint32_t plaintext_size)
+static uint32_t sgx_get_rsa_cipher_len(const uint32_t keyspec)
 {
-    if (aad_size > UINT32_MAX - sizeof(aes_gcm_key_data_t))
-        return UINT32_MAX;
-
-    if (plaintext_size > UINT32_MAX - sizeof(aes_gcm_key_data_t))
-        return UINT32_MAX;
-
-    if (aad_size > UINT32_MAX - plaintext_size)
-        return UINT32_MAX;
-
-    if (sizeof(aes_gcm_key_data_t) > UINT32_MAX - plaintext_size - aad_size)
-        return UINT32_MAX;
+    switch (keyspec) {
+        case EH_RSA_2048:
+            return RSA_OAEP_2048_CIPHER_LENGTH;
+        case EH_RSA_3072:
+            return RSA_OAEP_3072_CIPHER_LENGTH;
+        default:
+            return 0;
+    }
+}
 
-    return (aad_size + plaintext_size + sizeof(aes_gcm_key_data_t));
+static uint32_t sgx_get_rsa_padding_size(uint8_t padding_mode)
+{
+    switch (padding_mode) {
+        case RSA_PKCS1_OAEP_PADDING:
+            return 41;
+        case RSA_PKCS1_PADDING:
+            return 11;
+        default:
+            return 0;
+    }
 }
 
-/**
- * @brief Initialize the variables needed to generate the aes_gcm key
- * 
- * @param sgxStatus result for enclave function
- * @param cmk storage key information
- * @param paramJson Pass in the key parameter in the form of JSON string
- * @return sgx_status_t 
- */
-sgx_status_t create_aes_key(sgx_status_t* sgxStatus, ehsm_keyblob_t* cmk, JsonObj paramJson)
+static uint32_t sgx_get_rsa_max_encryption_size(ehsm_keyblob_t *cmk)
 {
-    uint32_t key_size;
-    if (cmk->keybloblen == 0)
-    {
-        //TODO: compute keybloblen
-        
-        switch (cmk->metadata.keyspec)
-        {
-            case EH_AES_GCM_128:
-                key_size = 16;
-                break;
-            case EH_AES_GCM_192:
-                key_size = 24;
-                break;
-            case EH_AES_GCM_256:
-                key_size = 32;
-                break;
-            default:
-                return SGX_ERROR_INVALID_PARAMETER;
-        }
-        cmk->keybloblen = sgx_calc_gcm_data_size(0, key_size);
-        return SGX_SUCCESS;
+    switch (cmk->metadata.keyspec) {
+        case EH_RSA_2048:
+            return RSA_OAEP_2048_CIPHER_LENGTH - sgx_get_rsa_padding_size(cmk->metadata.padding_mode);
```

Comment:
seems you already defined a function: sgx_get_rsa_cipher_len(), so does it need to use the switch..case here?

Reply:
There are some conflicts here. I hope to find the macro or function that has been defined. The three methods are reserved for the time being